### ISSUE TITLE
Value of CORS header is false

### ISSUE THREAD

**Issue by zyrouge** [fastify/fastify-cors#124]
## 🐛 Bug Report

Value of `access-control-allow-origin` is false when using RegExps and arrays. Works fine when using string though.

## To Reproduce

```js
await app.server.register(cors, {
    origin: ["http://localhost:3000"],
});
```

![image](https://user-images.githubusercontent.com/50770114/115705522-4433ec80-a38a-11eb-85bb-458c955963b5.png)
![image](https://user-images.githubusercontent.com/50770114/115705151-d4256680-a389-11eb-8a1a-e2cdcd26960a.png)

## Expected behavior

The header being set to the origin if allowed and not present if denied.

## Your Environment

- *node version*: 14.16.0
- *fastify version*: 3.12.0
- *os*: Windows

**Comment by mcollina** [comment]
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by zekth** [comment]
Could you please specify the version of `fastify-cors` you're using?
Also please provide the request headers.

**Comment by zyrouge** [comment]
> Could you please specify the version of `fastify-cors` you're using?
> Also please provide the request headers.
> 
> I cannot reproduce using:
> 
> ```
> fastify 3.15.0
> fastify-cors 5.2.0
> ```

It occurred while using v5.2.0, I'll try once again and try to identify the issue! :)

**Comment by Fdawgs** [comment]
@zyrouge do you have a public repo where this is occurring that you are able to share?

**Comment by zyrouge** [comment]
> @zyrouge do you have a public repo where this is occurring that you are able to share?

Wish i could but its private :(

**Comment by zyrouge** [comment]
I reproduced it, check https://github.com/zyrouge/fastify-cors/tree/header-false for the code.
![image](https://user-images.githubusercontent.com/50770114/116359215-2e0bad80-a81c-11eb-8e12-bce414a1407e.png)

**Comment by zyrouge** [comment]
Assuming that the `false` value comes from https://github.com/fastify/fastify-cors/blob/1385db3969dcc6d1b7326fce023a931ecde77948/index.js#L194 and browsers can't understand the `false` value and throws up a warning. According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin `null` value can be used but is not preferred, as a response with no header will be the solution?

**Comment by zekth** [comment]
`false` value means that the request is not allowed. Also you're not sharing the request `headers`, there is no way to reproduce it properly.

**Comment by zyrouge** [comment]
> `false` value means that the request is not allowed. Also you're not sharing the request `headers`, there is no way to reproduce it properly.

I did some research and found that the cors header should only have a value of `*` or an origin. I don't think `false` is the appropriate value to be set. It still does the job but it would be nice if there is a solution to get rid of warnings displayed in the browser when using it. I'll make sure to share the request headers. :)

**Comment by zekth** [comment]
This need to be adressed in another issue.

Is your issue solved?

**Comment by zyrouge** [comment]
> This need to be adressed in another issue.
> 
> Is your issue solved?

Fix for #127 would solve this.

**Comment by mcollina** [comment]
Fixed in v6.0.0
